runtime/Java/src/org/antlr/v4/runtime/atn/ParserATNSimulator.java

```diff
@@ -1552,6 +1552,55 @@ protected void closure_(ATNConfig config,
         }
 
 		for (int i=0; i<p.getNumberOfTransitions(); i++) {
+			// This block implements first-edge elimination of ambiguous LR
+			// alternatives as part of dynamic disambiguation during prediction.
+			// See antlr/antlr4#1398.
+			if (i == 0 && p.getStateType() == ATNState.STAR_LOOP_ENTRY && ((StarLoopEntryState)p).isPrecedenceDecision) {
+				boolean suppress = !config.context.isEmpty();
+				int limit = config.context.size();
+				if (config.context.hasEmptyPath()) {
```

Comment:
❗️ This is likely a bug. If we have an empty path we should automatically *not* suppress the edge.